src/main/java/com/lorepo/icplayer/client/model/Content.java

```diff
@@ -207,7 +207,9 @@ public String toXML(){
 		// Addons
 		xml += "<addons>";
 		for(AddonDescriptor descriptor : addonDescriptors.values()){
-			xml += "<addon-descriptor addonId='"+descriptor.getAddonId()+"' href='"+descriptor.getHref()+"'/>";
+		    if (descriptor.getAddonId() != "null") {
```

Comment:
Kila spraw:
1) Po pierwsze, takie porównanie nie zadziałałoby normalnie w javie, bo sprawdzane byłoby, czy wynik getAddonId() i "null" są tym samym obiektem. Ponieważ ten kod jest kompilowany do js'a możliwe że akurat w tym wypadku byłoby to ok, ale mimo wszystko dla pewności i przejrzystości kodu zamieniłbym to na descriptor.getAddonId().equals("null")
2) Wydaje mi się prawdopodobne, że getAddonId zwrócił oryginalnie nie stringa "null", tylko zwykłego nulla, który stringiem "null" stał się dopiero w wyniku pokręconych zasad rzutowania w js, więc dodałbym też warunek, który by to sprawdzał (tj. descriptor.getAddonId() == null).